### PR TITLE
feat: make semgrep reports more informative

### PR DESCRIPTION
**Pull request by art-tykh**
Add **code snippet** to finding desciption.

In new format description will look like this
image

### CODE REVIEW

**Comment by Maffooch** [comment]
Here is the failing snippet. Probably an unexpected value somewhere if I had to guess
```
2021-11-04T14:25:33.6480061Z [36muwsgi_1         |[0m ======================================================================
2021-11-04T14:25:33.6481346Z [36muwsgi_1         |[0m ERROR: test_parse_many_vulns (dojo.unittests.tools.test_semgrep_parser.TestSemgrepParser)
2021-11-04T14:25:33.6482739Z [36muwsgi_1         |[0m ----------------------------------------------------------------------
2021-11-04T14:25:33.6483647Z [36muwsgi_1         |[0m Traceback (most recent call last):
2021-11-04T14:25:33.6484839Z [36muwsgi_1         |[0m   File "/app/dojo/unittests/tools/test_semgrep_parser.py", line 84, in test_parse_many_vulns
2021-11-04T14:25:33.6486252Z [36muwsgi_1         |[0m     self.assertContains(" 'xsl-style-sheet': temp.name}", finding.description)
2021-11-04T14:25:33.6487796Z [36muwsgi_1         |[0m   File "/usr/local/lib/python3.8/site-packages/django/test/testcases.py", line 461, in assertContains
2021-11-04T14:25:33.6489044Z [36muwsgi_1         |[0m     text_repr, real_count, msg_prefix = self._assert_contains(
2021-11-04T14:25:33.6490332Z [36muwsgi_1         |[0m   File "/usr/local/lib/python3.8/site-packages/django/test/testcases.py", line 432, in _assert_contains
2021-11-04T14:25:33.6491505Z [36muwsgi_1         |[0m     response.status_code, status_code,
2021-11-04T14:25:33.6492530Z [36muwsgi_1         |[0m AttributeError: 'str' object has no attribute 'status_code'
2021-11-04T14:25:33.6493419Z [36muwsgi_1         |[0m 
2021-11-04T14:25:33.6494692Z [36muwsgi_1         |[0m ----------------------------------------------------------------------
```

**Comment by art-tykh** [comment]
@Maffooch I fixed the failing unit test. Probably `many_vulns.json` file was change in some other PR. I do not know why values were changed, but now everything works as expected.

File: dojo/tools/semgrep/parser.py [hunk 1]
```diff
@@ -71,3 +71,15 @@ def convert_severity(self, val):
...
         else:
             raise ValueError(f"Unknown value for severity: {val}")
+
+    def get_description(self, item):
+        description = ''
+
+        message = item["extra"]["message"]
+        description += '**Result message:** {}\n'.format(message)
+
+        snippet = item["extra"]["lines"]
```
Comment (StefanFl) [inline dojo/tools/semgrep/parser.py:81]:
```suggestion
        snippet = item["extra"].get("lines")
```

Using `get` will be safer, if `lines` shouldn't be in the report.

**Review by StefanFl — CHANGES_REQUESTED** [review]

File: dojo/tools/semgrep/parser.py (hunk 1 above)
Comment (art-tykh) [reply dojo/tools/semgrep/parser.py]: Fixed, thanks for the suggestion.

**Review by StefanFl — CHANGES_REQUESTED** [review]
Thanks for your change, that's good. If you would change one of the existing unit test cases, so that it checks the new format of the description, then I can approve your pull request.

**Comment by art-tykh** [comment]
@StefanFl  I've added test for new description format. Job with unittests finished successful, but job with integration tests failed with
```
integration-tests_1  | ======================================================================
integration-tests_1  | FAIL: test_delete_finding (__main__.FindingTest)
integration-tests_1  | ----------------------------------------------------------------------
integration-tests_1  | Traceback (most recent call last):
integration-tests_1  |   File "tests/finding_test.py", line 493, in test_delete_finding
integration-tests_1  |     self.assertTrue(self.is_success_message_present(text='Finding deleted successfully'))
integration-tests_1  | AssertionError: False is not true
integration-tests_1  | 
integration-tests_1  | ----------------------------------------------------------------------
integration-tests_1  | Ran 30 tests in 104.910s
integration-tests_1  | 
integration-tests_1  | FAILED (failures=1)
```

Could, you, please, advice what I am doing wrong? I didn't touch functionality connected with finding deletion.

**Comment by StefanFl** [comment]
You haven't done anything wrong. The integration tests on GitHub fail from time to time. It's very random. Running them again often helps

**Review by StefanFl — APPROVED** [review]

**Review by Maffooch — APPROVED** [review]

**Pull request merged**